internal/provider/provider.go

```diff
@@ -0,0 +1,818 @@
+package provider
```

Comment:
I believe that this file mimics the current provider config.

Is something worth mentioning here which is an additional thing? (we can talk about it on Monday).

Reply:
The Configure() function is required as part of the interface of being a provider. You could introduce other design patterns such as builder as part of that, but you still need this function.

And you are right, it does match the current provider config. For good reason, it has to for the provider muxing to work. If they didn't match exactly then we would not be able to write resources using the plugin framework.